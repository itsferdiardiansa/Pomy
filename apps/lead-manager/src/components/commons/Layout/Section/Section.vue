<template>
  <div :class="`${prefixClass}-section`">
    <div :class="`${prefixClass}-section--title`" v-if="title">
      <h3>{{ title }}</h3>
    </div>

    <div :class="`${prefixClass}-section--content`">
      <slot name="content" />
    </div>
  </div>
</template>
<script>
export default {
  props: {
    title: {
      type: String,
      default: '  '
    }
  }
}
</script>
<style lang="scss" scoped>
.#{$prefixClass}-section {
  @apply pb-8 flex flex-col my-2;

  &--title {
    @apply py-6;

    h3 {
      @apply font-black text-2xl;
    }
  }

  &--content {
    @apply relative;
  }
}
</style>
