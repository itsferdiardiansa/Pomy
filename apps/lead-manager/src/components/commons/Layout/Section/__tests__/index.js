import { mount } from '@vue/test-utils'
import Section from '@/components/commons/Layout/Section'

describe('common/Layout/Section', () => {
  it('will render and match snapshot', () => {
    const Component = mount(<Section title="Title" />)

    expect(Component.element).toMatchSnapshot()
  })
})
