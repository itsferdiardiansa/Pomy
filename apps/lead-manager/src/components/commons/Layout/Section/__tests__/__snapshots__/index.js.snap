// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Layout/Section will render and match snapshot 1`] = `
<div
  class="ss-section"
>
  <div
    class="ss-section--title"
  >
    <h3>
      Title
    </h3>
  </div>
  <div
    class="ss-section--content"
  >
    
    
  </div>
</div>
`;
